src/views/AddItem.jsx

```diff
@@ -1,25 +1,44 @@
 import { useState } from 'react';
 import { addItem } from '../api/firebase.js';
 
-export function AddItem({ listToken }) {
+export function AddItem({ listToken, data }) {
 	const [itemName, setItemName] = useState('');
 	const [daysUntilNextPurchase, setDaysUntilNextPurchase] = useState(7);
-	const [itemAdded, setItemAdded] = useState(false);
 	const [error, setError] = useState(false);
+	const [userAlertMessage, setUserAlertMessage] = useState('');
 
-	// TODO: dynamically pass in the list token so item is added to correct list
 	const handleSubmit = async (e) => {
 		e.preventDefault();
+		const normalizedItemName = itemName.replace(/[\s\W]|_+/g, '').toLowerCase();
+
+		if (normalizedItemName === '') {
+			setUserAlertMessage('Please enter an item name.');
+			return;
+		}
+
+		for (const item of data) {
```

Comment:
Am I the only one that has to google '`for...in` vs `for...of`' whenever I have to use them? ._.

Reply:
lol nope!

Reply:
This was one of those that hit me just before falling asleep that night. "Oh yeah, we need to use a for loop" ZZZzzzzzz.....